components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -378,6 +380,47 @@ public MLModel retrieveModel(long modelId) throws MLModelBuilderException {
         }
     }
 
+    /**
+     * Publish a ML model to registry.
+     *
+     * @param tenantId    Unique ID of the tenant.
+     * @param userName    Username of the user.
+     * @param modelId     Unique ID of the built ML model.
+     * @throws MLModelPublisherException
+     */
+    public void publishModel(int tenantId, String userName, long modelId) throws MLModelPublisherException {
+        InputStream in = null;
+        try {
+            // read model
+            MLStorage storage = databaseService.getModelStorage(modelId);
+            String storageType = storage.getType();
+            String storageLocation = storage.getLocation();
+            MLIOFactory ioFactory = new MLIOFactory(mlProperties);
+            MLInputAdapter inputAdapter = ioFactory.getInputAdapter(storageType + MLConstants.IN_SUFFIX);
+            in = inputAdapter.read(new URI(storageLocation));
+            if (in == null) {
+                throw new MLModelPublisherException("Invalid model [id] " + modelId);
+            }
+            // create registry path
+            MLCoreServiceValueHolder valueHolder = MLCoreServiceValueHolder.getInstance();
+            String modelName = databaseService.getModel(tenantId, userName, modelId).getName();
+            String registryPath = File.separator + valueHolder.getModelRegistryLocation() + File.separator + modelName;
+            // publish to registry
+            RegistryOutputAdapter registryOutputAdapter = new RegistryOutputAdapter();
+            registryOutputAdapter.write(registryPath, in);
+
+        } catch (Exception e) {
```

Comment:
This is wrong, if the specific block can throw multiple checked exceptions you need to handle them properly, catching "Exception" is very lazy anti-pattern and we don't allow to catch "Exception" unless there is a special and already discussed reason to do so.